src/theory/arith/nl/iand_solver.cpp

```diff
@@ -264,8 +264,9 @@ Node IAndSolver::sumBasedLemma(Node i)
   Assert(i.getKind() == IAND);
   Node x = i[0];
   Node y = i[1];
-  size_t bvsize = i.getOperator().getConst<IntAnd>().d_size;
-  uint64_t granularity = options().smt.BVAndIntegerGranularity;
+  uint32_t bvsize = i.getOperator().getConst<IntAnd>().d_size;
+  Assert(options().smt.BVAndIntegerGranularity <= 8);
+  uint32_t granularity = (uint32_t)options().smt.BVAndIntegerGranularity;
```

Comment:
Use c++ casts:
```suggestion
  uint32_t granularity = static_cast<uint32_t>(options().smt.BVAndIntegerGranularity);
```
They are checked at compile time (c-style casts are not).